### ISSUE TITLE
toJson properties are not mapped from implemented

### ISSUE THREAD

**Issue by Noteofman** [heftapp/graphql_codegen#192]
In the latest version 0.11.1, based upon a GQL query with multiple property implementations, the toJson method does not seem to be generating the properties on the implemented class, in the below example props1, 2 and 3.

```
  final String $__typename;

  final double? prop1;

  final double? prop2;

  final Currency currency;

  final double? prop3;

  final double fixed;

  final double? discount;

  final String? discountFootnote;

  Map<String, dynamic> toJson() {
    final _resultData = <String, dynamic>{};
    final l$$__typename = $__typename;
    _resultData['__typename'] = l$$__typename;
    final l$fixed = fixed;
    _resultData['fixed'] = l$fixed;
    final l$discount = discount;
    _resultData['discount'] = l$discount;
    final l$discountFootnote = discountFootnote;
    _resultData['discountFootnote'] = l$discountFootnote;
    return _resultData;
  }
```

**Comment by budde377** [comment]
Can you please provide a reproducible example

**Comment by Noteofman** [comment]
> Can you please provide a reproducible example

@budde377 Check out the following project, specifically the generated toJson methods:

https://github.com/Noteofman/graphql_codegen_bug/blob/master/lib/test.graphql.dart

E.g the class `Query$Testing$books$$Textbook`

This seems to be as a result of including the properties in the parent block in the query rather than inside the interface scope. As in

   ```
    books {
        title
        author
        ... on Textbook {
            courses
        }

        ... on ColoringBook {
            colors
        }
    }
```

You can see not all the properties are mapped, whereas they are in the fromJson methods

**Comment by budde377** [comment]
Thanks for reporting this and providing an example. It should now be fixed in `0.11.2`.
